src/Transport/AmqpTransport.php

```diff
@@ -29,6 +31,17 @@ public function keepalive(Envelope $envelope, ?int $seconds = null): void
 
             $this->getMessageCount();
         } catch (\Throwable $throwable) {
+            $previous = $this->getTransportException($throwable);
+            if (null !== $previous) {
+                /*
+                 * TransportException like AMQPQueueException("Server channel error: 406, message: PRECONDITION_FAILED - delivery acknowledgement on channel 1 timed out."),
+                 * connection lost
+                 * message will be Redelivered
+                 * need to stop worker and prevent for requeue current message
+                 */
+                throw new UnrecoverableMessageHandlingException($previous->getMessage(), $previous->getCode(), $previous);
```

Comment:
won't we lose some information by skipping current throwable? 
why is it bad idea to have it nested by passing current throwable as is?

Reply:
Because this exception goes to message header. It nice to have real exception and exception message in this specific problem.
When you run `messenger:failed:show id` you will get real cause of this error.